### PR TITLE
replace user id with displayname and add type to diffs

### PR DESCRIPTION
**Pull request by chagai95**
Fixes https://github.com/vector-im/element-web/issues/3969

Makes more sense to show the display name and not the user ID, I think.

Translation string needs to be changed for this as well:
https://translate.element.io/translate/element-web/matrix-react-sdk/en/?checksum=5db8fd2d99ec50a6

I could not figure it out, or I don't have the rights to change it.
Before:
image
After:
image

Signed-off-by: Chagai Friedlander chagai95@gmail.com

---
Here's what your changelog entry will look like:

## ✨ Features
 * replace user id with displayname and add type to diffs ([\#7781](https://github.com/matrix-org/matrix-react-sdk/pull/7781)). Fixes vector-im/element-web#3969. Contributed by @chagai95.

Preview: https://pr7781--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/TextForEvent.tsx [hunk 1]
```diff
@@ -498,10 +498,10 @@ function textForPowerEvent(event: MatrixEvent): () => string | null {
         if (!Number.isInteger(to)) {
             to = currentUserDefault;
         }
+        const displayName = MatrixClientPeg.get().getUser(userId).displayName;
```
Comment (t3chguy) [inline src/TextForEvent.tsx:501]: Displaynames are per-room, not global, so you can't use the displayName of getUser in a room. You'll need to get the room member instead

File: src/TextForEvent.tsx [hunk 2]
```diff
@@ -512,8 +512,8 @@ function textForPowerEvent(event: MatrixEvent): () => string | null {
     return () => _t('%(senderName)s changed the power level of %(powerLevelDiffText)s.', {
         senderName,
         powerLevelDiffText: diffs.map(diff =>
-            _t('%(userId)s from %(fromPowerLevel)s to %(toPowerLevel)s', {
-                userId: diff.name,
+            _t('%(displayName)s from %(fromPowerLevel)s to %(toPowerLevel)s', {
+                displayName: diff.displayName,
```
Comment (t3chguy) [inline src/TextForEvent.tsx:516]: You need to run the `i18n` script to regen the i18n files

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Comment by t3chguy** [comment]
Hello @chagai95 are you able to iterate on this PR based on the feedback?

Thanks

**Comment by chagai95** [comment]
> Hello @chagai95 are you able to iterate on this PR based on the feedback?
> 
> Thanks

Hey, I should be able to, thx for the ping, I'll try getting around to this very soon.

**Comment by aaronraimist** [comment]
Superseded by https://github.com/matrix-org/matrix-react-sdk/pull/9295

**Comment by turt2live** [comment]
Closing in favour of https://github.com/matrix-org/matrix-react-sdk/pull/9295

**Pull request closed without merging**